backend/src/routers/hint_router.py

```diff
@@ -0,0 +1,14 @@
+from fastapi import APIRouter, Depends
+from ..models.hint_request import HintRequest
+from ..services.langchain_service import get_hint_chain
+
+hint_router = APIRouter()
+
+@hint_router.get("/")
+async def root() -> str:
+    return "Welcome to LeetCopilot!"
+
+@hint_router.post("/hint", response_model=str)
+async def get_hint(promptData: HintRequest):
```

Comment:
Should we be setting the return type of this to `HintResponse`?

Reply:
Yes